packages/frontend/src/app/repository/tests-list/test-details/test-details.component.ts

```diff
@@ -14,17 +14,37 @@
 
 import {Component, Input} from '@angular/core';
 import {Test} from 'src/app/services/search/interfaces';
+import {COMService} from '../../../services/com/com.service';
+import {Observable} from 'rxjs';
 
 @Component({
   selector: 'app-test-details',
   templateUrl: './test-details.component.html',
   styleUrls: ['./test-details.component.css'],
 })
 export class TestDetailsComponent {
+  constructor(public comService: COMService) {}
   @Input() test: Test;
+  @Input() repoName: string;
+  @Input() orgName: string;
+  windowProvider = window;
+  public deleteUrl$: Observable<string>;
 
   //Converts tests' passing percentage from decimal to percentage
   toPercentage(percentpassing: number): string {
     return (percentpassing * 100).toFixed(2);
   }
+
+  startDeleteTest() {
+    this.deleteUrl$ = this.comService.fetchDeleteTestUrl(
+      this.orgName,
+      this.repoName,
+      this.test.name,
+      'https://flaky-dashboard.web.app/'
+    );
+    this.deleteUrl$.subscribe(res => {
+      console.info(res);
+      this.windowProvider.open(res, '_self');
+    });
+  }
 }
```

Comment:
Since `deleteUrl$` is only used in `startDeleteTest`, it probably doesn't need to be an attribute of the component?
Also the observable can be immediately used without storing it at all. Like:
```
this.comService.fetchDeleteTestUrl(
 this.orgName,
 this.repoName,
 this.test.name,
 'https://flaky-dashboard.web.app/'
).subscribe(res => {
 this.windowProvider.open(res, '_self');
});
```

Reply:
Oh perfect, thank you!